engine/src/main/java/org/terasology/world/block/family/MultiConnectFamily.java

```diff
@@ -36,160 +34,62 @@
-            .put(SIX_CONNECTIONS, (byte) 63)
-            .build();
+    protected TByteObjectMap<Block> blocks = new TByteObjectHashMap<>();
     
     public MultiConnectFamily(BlockFamilyDefinition definition, BlockShape shape, BlockBuilderHelper blockBuilder) {
         super(definition, shape, blockBuilder);
+        this.setBlockUri(new BlockUri(definition.getUrn()));
+        this.setCategory(definition.getCategories());
     }
 
     public MultiConnectFamily(BlockFamilyDefinition definition, BlockBuilderHelper blockBuilder) {
         super(definition, blockBuilder);
-        TByteObjectMap<String>[] basicBlocks = new TByteObjectMap[7];
-        blocks = new TByteObjectHashMap<>();
-
-        this.registerConnections(basicBlocks);
-
-        BlockUri blockUri = new BlockUri(definition.getUrn());
-
-        // Now make sure we have all combinations based on the basic set (above) and rotations
-        for (byte connections = 0; connections < 64; connections++) {
-            // Only the allowed connections should be created
-            if ((connections & getConnectionSides()) == connections) {
-                Block block = constructBlockForConnections(connections, blockBuilder, definition, basicBlocks);
-                if (block == null) {
-                    throw new IllegalStateException("Unable to find correct block definition for connections: " + connections);
-                }
-                block.setUri(new BlockUri(blockUri, new Name(String.valueOf(connections))));
-                block.setBlockFamily(this);
-                blocks.put(connections, block);
-            }
-        }
-        this.archetypeBlock = blocks.get(SideBitFlag.getSides(Side.RIGHT, Side.LEFT));
-        this.setBlockUri(blockUri);
+        this.setBlockUri(new BlockUri(definition.getUrn()));
         this.setCategory(definition.getCategories());
-    }
 
-    public byte getConnectionSides() {
-        return 64;
     }
 
+    protected abstract boolean connectionCondition(Vector3i blockLocation, Side connectSide);
 
-    protected void registerConnections(TByteObjectMap<String>[] basicBlocks) {
-        addConnections(basicBlocks, 0, NO_CONNECTIONS);
-        addConnections(basicBlocks, 1, ONE_CONNECTION);
-        addConnections(basicBlocks, 2, TWO_CONNECTIONS_LINE);
-        addConnections(basicBlocks, 2, TWO_CONNECTIONS_CORNER);
-        addConnections(basicBlocks, 3, THREE_CONNECTIONS_CORNER);
-        addConnections(basicBlocks, 3, THREE_CONNECTIONS_T);
-        addConnections(basicBlocks, 4, FOUR_CONNECTIONS_CROSS);
-        addConnections(basicBlocks, 4, FOUR_CONNECTIONS_SIDE);
-        addConnections(basicBlocks, 5, FIVE_CONNECTIONS);
-        addConnections(basicBlocks, 6, SIX_CONNECTIONS);
-    }
-
+    public abstract byte getConnectionSides();
 
-    public boolean horizontalOnly() {
-        return false;
-    }
+    public abstract boolean horizontalOnly();
```

Comment:
Doesn't look like this is used (at least in this or the example code). Can it be removed? If not, is there reason for not having verticalOnly?

Feels like a leftover or a weird thing used by something else in the engine. In that case I'd like to see something like `shouldRotateVertically` or whatever the appropriate predicate is so that implementations of this class can own the logic. That may be unclear, but as an example `animal.canSwim()` instead of `animal.isFish()`

Reply:
There is a lot of legacy stuff I'm unsure about.